Cubical/Categories/Instances/FPAlgebrasSmall.agda

```diff
@@ -0,0 +1,334 @@
+    CommAlgebraHom (indFPAlg A) (indFPAlg B)
+  id FPAlgebrasSmallCategory = idCommAlgebraHom _
+  _⋆c_ FPAlgebrasSmallCategory = compCommAlgebraHom _ _ _
+  ⋆IdL FPAlgebrasSmallCategory = compIdCommAlgebraHom
+  ⋆IdR FPAlgebrasSmallCategory = idCompCommAlgebraHom
+  ⋆Assoc FPAlgebrasSmallCategory = compAssocCommAlgebraHom
+  isSetHom FPAlgebrasSmallCategory = isSetAlgebraHom _ _
+
+
+  -- the name of this module will be justified once we have defined
+  -- Zariski sheaves on small f.p. algebras
+  module SchemesOfFinitePresentation where
+
+    -- "parsimonious" presheaf categories
+    fpAlgFunctor = Functor FPAlgebrasSmallCategory (SET ℓ)
+    fpAlgFUNCTOR = FUNCTOR FPAlgebrasSmallCategory (SET ℓ)
+
+    -- yoneda in the notation of Demazure-Gabriel
+    -- uses η-equality of Categories
+    Sp : Functor (FPAlgebrasSmallCategory ^op) fpAlgFUNCTOR
+    Sp = YO {C = (FPAlgebrasSmallCategory ^op)}
+
+    -- the Zariski lattice (classifying compact open subobjects)
+    private forget = ForgetfulCommAlgebra→CommRing R
+
+    𝓛 : fpAlgFunctor
+    F-ob 𝓛 A = ZL (forget .F-ob (indFPAlg A)) , SQ.squash/
+    F-hom 𝓛  φ = inducedZarLatHom (forget .F-hom φ) .fst
+    F-id 𝓛 = cong (λ x → inducedZarLatHom x .fst) (forget .F-id)
+            ∙ cong fst (inducedZarLatHomId _)
+    F-seq 𝓛 _ _ = cong (λ x → inducedZarLatHom x .fst) (forget .F-seq _ _)
+                 ∙ cong fst (inducedZarLatHomSeq _ _)
+
+    CompactOpen : fpAlgFunctor → Type ℓ
+    CompactOpen X = X ⇒ 𝓛
+
+    isAffine : fpAlgFunctor → Type ℓ
+    isAffine X = ∃[ A ∈ SmallFPAlgebra ] (Sp .F-ob A) ≅ᶜ X
+
+    -- the induced subfunctor
+    ⟦_⟧ᶜᵒ : {X : fpAlgFunctor} (U : CompactOpen X) → fpAlgFunctor
+    F-ob (⟦_⟧ᶜᵒ {X = X} U) A = (Σ[ x ∈ X .F-ob A .fst  ] U .N-ob A x ≡ D _ 1r)
+                                  , isSetΣSndProp (X .F-ob A .snd) λ _ → squash/ _ _
+     where instance _ = snd (forget .F-ob (indFPAlg A))
+    F-hom (⟦_⟧ᶜᵒ {X = X} U) {x = A} {y = B} φ (x , Ux≡D1) = (X .F-hom φ x) , path
+      where
+      instance
+        _ = (forget .F-ob (indFPAlg A)) .snd
+        _ = (forget .F-ob (indFPAlg B)) .snd
+      open IsLatticeHom
+      path : U .N-ob B (X .F-hom φ x) ≡ D _ 1r
+      path = U .N-ob B (X .F-hom φ x) ≡⟨ funExt⁻ (U .N-hom φ) x ⟩
+             𝓛 .F-hom φ (U .N-ob A x) ≡⟨ cong (𝓛 .F-hom φ) Ux≡D1 ⟩
+             𝓛 .F-hom φ (D _ 1r)      ≡⟨ inducedZarLatHom (forget .F-hom φ) .snd .pres1 ⟩
+             D _ 1r ∎
+    F-id (⟦_⟧ᶜᵒ {X = X} U) = funExt (λ x → Σ≡Prop (λ _ → squash/ _ _)
+                                       (funExt⁻ (X .F-id) (x .fst)))
+    F-seq (⟦_⟧ᶜᵒ {X = X} U) φ ψ = funExt (λ x → Σ≡Prop (λ _ → squash/ _ _)
+                                            (funExt⁻ (X .F-seq φ ψ) (x .fst)))
+
+    isAffineCompactOpen : {X : fpAlgFunctor} (U : CompactOpen X) → Type ℓ
+    isAffineCompactOpen U = isAffine ⟦ U ⟧ᶜᵒ
+
+    -- the dist. lattice of compact opens
+    CompOpenDistLattice : Functor fpAlgFUNCTOR (DistLatticesCategory {ℓ} ^op)
```

Comment:
"Homework" question for us: Can this be unified with the very similar construction of the "global sections functor" that maps `X` to `X ⇒ 𝔸¹`?